pkg/cli/admin/top/persistentvolumeclaims.go

```diff
@@ -0,0 +1,382 @@
+		return persistentVolumeClaimsBytes, fmt.Errorf("metrics not available yet")
+	}
+
+	return persistentVolumeClaimsBytes, nil
+}
+
+// getWithBearer gets a Route by namespace/name, constructs a URI using
+// status.ingress[].host and the path argument, and performs GETs on that
+// URI using Bearer authentication with the token argument.
+func getWithBearer(ctx context.Context, getRoute RouteGetter, namespace, name string, baseURI *url.URL, bearerToken string, InsecureTLS bool) ([]byte, error) {
+	if len(bearerToken) == 0 {
+		return nil, fmt.Errorf("no token is currently in use for this session")
+	}
+
+	route, err := getRoute(ctx, namespace, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	httpTransport := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: InsecureTLS},
+	}
+
+	withDebugWrappers, err := transport.HTTPWrappersForConfig(
+		&transport.Config{
+			UserAgent:   rest.DefaultKubernetesUserAgent() + "(top persistentvolumeclaims)",
+			BearerToken: bearerToken,
+		},
+		httpTransport,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{Transport: withDebugWrappers}
+	errs := make([]error, 0, len(route.Status.Ingress))
+	for _, ingress := range route.Status.Ingress {
+		uri := *baseURI
+		uri.Host = ingress.Host
+		content, err := checkedGet(uri, client)
+		if err == nil {
+			return content, nil
+		} else {
+			errs = append(errs, fmt.Errorf("%s->%w", ingress.Host, err))
+		}
+	}
+
+	if len(errs) == 1 {
+		return nil, fmt.Errorf("unable to get %s from URI in the %s/%s Route: %s", baseURI.Path, namespace, name, errors.NewAggregate(errs))
+	} else {
+		return nil, fmt.Errorf("unable to get %s from any of %d URIs in the %s/%s Route: %s", baseURI.Path, len(errs), namespace, name, errors.NewAggregate(errs))
+	}
+
+}
+
+func checkedGet(uri url.URL, client *http.Client) ([]byte, error) {
+	req, err := http.NewRequest("GET", uri.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	glogBody("Response Body", body)
+
+	if resp.StatusCode != http.StatusOK {
+		return body, fmt.Errorf("GET status code=%d", resp.StatusCode)
+	}
+
+	return body, nil
+}
+
+type PromOutput struct {
+	Status string         `json:"status"`
+	Data   PromOutputData `json:"data"`
+}
+
+type PromOutputData struct {
+	ResultType string                 `json:"resultType"`
+	Result     []PromOutputDataResult `json:"result"`
+}
+
+type PromOutputDataResult struct {
+	Metric map[string]string `json:"metric"`
+	Value  FloatStringPair   `json:"value"`
+}
+
+type FloatStringPair [2]interface{}
+
+// glogBody and truncateBody taken from client-go Request
+// https://github.com/openshift/oc/blob/4be3c8609f101a8c5867abc47bda33caae629113/vendor/k8s.io/client-go/rest/request.go#L1183-L1215
```

Comment:
We need it here to show the original response body from Prometheus which may be useful for troubleshooting purpouses.

We can't use it from `client-go`, because here we are not executing a REST API call to kuberentes using `client-go`, but instead we are executing a REST API call to Prometheus endpoint (for API calls to kuberentes using `client-go` would have been superfluous as the code already incorporated by default, see [here](https://github.com/kubernetes/client-go/blob/release-1.30/rest/request.go#L1202-L1215)).